atoma-proxy/src/server/middleware.rs

```diff
@@ -457,6 +460,41 @@ pub async fn confidential_compute_middleware(
     Ok(next.run(req).await)
 }
 
+/// Middleware to track request latency and middleware timing
+#[instrument(skip_all)]
+pub async fn track_metrics(request: Request<Body>, next: Next) -> Result<Response> {
+    let start = Instant::now();
```

Comment:
Duration time should be split across successful and failed requests. I suggest tracking as well the number of failed vs successful requests as part of the metrics pipeline

Reply:
Also from a statistically point of view, this seems to be incoherent, as we are storing all these metrics as a part of a single distribution that is common to all endpoints, when this is not the case. Chat completions vs node creation vs image gen have different distributions, and therefore we should collect metrics per available `handler`. Unless, I am actually wrong in thinking this, and open-telemetry already decouples the metric collection for different handler endpoints (which I would be surprised as it does not so happen with the `prometheus` library)